iac/network/scripts/deploy_bootnodes.sh

```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+
+set -e
+
+# This script should be run from the root of iac/network. It will walk through the process of deploying a set of bootnodes. To do this, it will
+# 1. Create an SSH key and store i a GCP secret
+# 2. Create a service account in GCP and appropriate firewall rules for running bootnodes
+# 3. Create a static IP address for each provided region
+# 4. Create a P2P private key for each region if one doesn't already exist, this is stored as a GCP secret
+# 5. Generate the ENRs for the IP/Private key pairs
+# 6. Writes the ENRs to the provided S3 bucket
+# 7. Creates a VM in each region of the provided machine type
+# 8. Executes a startup script and updates systemd to ensure the bootnode is always running
+
+# Usage: ./scripts/deploy_bootnodes.sh <network-name> "region1,region2" "t2d-standard-2" "s3://static.aztec.network" <L1-chain-id> <gcp-project-id>
+
+NETWORK_NAME=${1:-}
+GCP_REGIONS=${2:-}
+GCP_MACHINE_TYPE=${3:-}
+STATIC_S3_BUCKET=${4:-}
+L1_CHAIN_ID=${5:-}
+PROJECT_ID=${6:-}
+TAG=${7:-"latest"}
+
+P2P_UDP_PORT=40400
+P2P_TCP_PORT=40400
+P2P_UDP_PORTS="[\"$P2P_UDP_PORT\"]"
+P2P_TCP_PORTS="[\"$P2P_TCP_PORT\"]"
+
+echo "NETWORK_NAME: $NETWORK_NAME"
+echo "GCP_REGIONS: $GCP_REGIONS"
+
+if [[ -z "$NETWORK_NAME" ]]; then
+    echo "NETWORK_NAME is required"
+    exit 1
+fi
+
+if [[ -z "$STATIC_S3_BUCKET" ]]; then
+    echo "STATIC_S3_BUCKET is required"
+    exit 1
+fi
+
+if [[ -z "$PROJECT_ID" ]]; then
+    echo "PROJECT_ID is required"
+    exit 1
+fi
+
+ROOT=$PWD
```

Comment:
I'd change this for the `$(git rev-parse --show-toplevel)` trick for resolving to the repo root, so we don't risk issues if this script is run from elsewhere in the repo.

Reply:
Done